### DISCUSSION TITLE
Trouble with linking sibling modules (vs2019)

### DISCUSSION THREAD

**Discussion by thaellin** [premake/premake-core#1632] [Q&A] [answered]
I'm just getting started with premake, and am using the prebuilt distribution for premake5.0.0 alpha16 on Windows10 (home).  I have a test folder/project in which I am trying to work out how premake works.  

I find that external libs work fine in the links, (i.,e.: links({"opengl32"}) shows in the project lib inputs); however, linking another workspace project does NOT emit an additional lib input.  In particular:

workspace("test")
  project("ProjectA")
  language("C++")
  kind("StaticLib")
  files(...)
  ...
  project("ProjectB")
  language("C++")  
  kind("SharedLib")
  file(...)
  links({"ProjectA","externLib"})
  ...
  project("ProjectC")
  language("C++")  
  kind("ConsoleApp")
  files(...)
  links("ProjectA","ProjectB","anotherLib"})
  
...results in ProjectA as expected, ProjectB having only 'externLib' as a link dependency, and ProjectC having only 'anotherLib'.  My test also has 'dependson' which does generate the expected build order.  I have also removed dependson to be sure it was not interfering, and removed only the links in case dependson was meant to generate things on my behalf.

I assume I am doing something wrong, but I cannot find anything in the docs or online suggesting what it is.  I'd appreciate any guidance, since this is holding me back from being able to use premake (which has, otherwise, been an exciting tool to have discovered).  Premake obviously knows these projects are in the workspace and is treating them specially, since it would otherwise have just added them to the link libs.

I've zipped my actual current testbed configuration and attached it to this post.  In it, I have "DllProject" specifying a link to "LibProject" - curiously, it behaves as if no link libs were specified at all(default dll libs are listed in the project link list).  

I have tried links({"LibProject"}), links("LibProject") and links "LibProject" (yeah, I know) all with the same effect.  Also tried moving the test folder to my Document folder and a mapped drive with a fairly short path.  I'm stumped and would be happy if this is something stupid I am overlooking.

Thank you for any help you might provide.
-- Jeff Duncan

[testbed.zip](https://github.com/premake/premake-core/files/6563498/testbed.zip)

**Comment by samsinsane** [comment, marked as answer]
Sibling projects are linked via References rather than the additional libraries system. You should see them in the solution explorer view, rather than the project properties view.

**Reply by thaellin** [reply]
I'd never used visual studio's project references in a C++ project before (only C#), so I had not looked there...

The sibling is indeed referenced and if I had done a test build instead of manually checking everything was where I expected it to be then I would have found everything linked and works fine.

I knew this was going to be a stupid user error, but I had no idea /how/ stupid...

Thank you!

**Comment by starkos** [comment]
There is a *lot* going on in that example you sent. Are you still able to reproduce the issue with a _simple_ solution that just sets up the links? Premake wouldn't be very usable if it wasn't able to link to other projects; most likely there is an issue within your script itself.
